llo/stream_calculated.go

```diff
@@ -0,0 +1,435 @@
+		return decimal.Decimal{}, fmt.Errorf("precision is too large")
+	}
+	ad, err := toDecimal(x)
+	if err != nil {
+		return decimal.Decimal{}, err
+	}
+	return ad.Round(int32(precision)), nil
+}
+
+// toDecimal converts x to a decimal.Decimal
+func toDecimal(x any) (decimal.Decimal, error) {
+	switch v := x.(type) {
+	case string:
+		return decimal.NewFromString(v)
+	case int:
+		return decimal.NewFromInt(int64(v)), nil
+	case int32:
+		return decimal.NewFromInt32(v), nil
+	case int64:
+		return decimal.NewFromInt(v), nil
+	case float32:
+		return decimal.NewFromFloat32(v), nil
+	case float64:
+		return decimal.NewFromFloat(v), nil
+	case uint:
+		return decimal.NewFromUint64(uint64(v)), nil
+	case uint32:
+		return decimal.NewFromUint64(uint64(v)), nil
+	case uint64:
+		return decimal.NewFromUint64(v), nil
+	case decimal.Decimal:
+		return v, nil
+	default:
+		return decimal.Decimal{}, fmt.Errorf("unsupported type: %T", x)
+	}
+}
+
+// evalDecimal evaluates the given expression and returns the result as a decimal.Decimal
+func evalDecimal(stmt string, env environment) (decimal.Decimal, error) {
+	r, err := expr.Eval(stmt, env)
+	if err != nil {
+		return decimal.Decimal{}, fmt.Errorf("failed to evaluate expression: %w", err)
+	}
+
+	d, ok := r.(decimal.Decimal)
+	if !ok {
+		return decimal.Decimal{}, fmt.Errorf("expected decimal.Decimal, got %T", r)
+	}
+
+	return d, nil
+}
+
+// ProcessStreamCalculated evaluates the given expression and returns the result as a decimal.Decimal
+func (p *Plugin) ProcessStreamCalculated(outcome *Outcome) {
+	for cid, cd := range outcome.ChannelDefinitions {
+		if cd.ReportFormat == llotypes.ReportFormatEVMABIEncodeUnpackedExpr {
+			env := NewEnv()
+			for _, stream := range cd.Streams {
+				p.Logger.Debugw("setting stream value", "channelID", cid, "streamID", stream.StreamID, "aggregator", stream.Aggregator)
+				env.SetStreamValue(stream.StreamID, outcome.StreamAggregates[stream.StreamID][stream.Aggregator])
+			}
+
+			copt := opts{}
+			if err := json.Unmarshal(cd.Opts, &copt); err != nil {
+				p.Logger.Warnw("failed to unmarshal channel definition options", "channelID", cid, "error", err)
+				continue
+			}
+
+			for _, abi := range copt.ABI {
+				if abi.ExpressionStreamID == 0 {
+					p.Logger.Warnw("expression stream ID is 0", "channelID", cid, "expression", abi.Expression)
+					continue
+				}
+
+				if abi.Expression == "" {
+					p.Logger.Warnw("expression is empty", "channelID", cid, "expressionStreamID", abi.ExpressionStreamID)
+					continue
+				}
+
+				p.Logger.Infow("evaluating expression", "channelID", cid, "expression", abi.Expression, "env", fmt.Sprintf("%+v", env))
+				value, err := evalDecimal(abi.Expression, env)
+				if err != nil {
+					p.Logger.Warnw("failed to evaluate expression", "channelID", cid, "expression", abi.Expression, "error", err)
+					continue
+				}
+
+				outcome.StreamAggregates[abi.ExpressionStreamID] = map[llotypes.Aggregator]StreamValue{
```

Comment:
@brunotm what if somehow this streamID is already used - it will override it in this stream aggregates

should we check against that?